engine/config_integrations/alertmanager.py

```diff
@@ -22,26 +22,27 @@
 
 # Web
 web_title = """\
-{%- set groupLabels = payload.groupLabels.copy() -%}
-{%- set alertname = groupLabels.pop('alertname') | default("") -%}
+{%- set groupLabels = payload.get("groupLabels", {}).copy() -%}
+{%- set alertname = groupLabels.pop('alertname', "")  -%}
 
 
-[{{ payload.status }}{% if payload.status == 'firing' %}:{{ payload.numFiring }}{% endif %}] {{ alertname }} {% if groupLabels | length > 0 %}({{ groupLabels|join(", ") }}){% endif %}
-"""  # noqa
+[{{ payload.status }}{% if payload.status == 'firing' and payload.numFiring %}:{{ payload.numFiring }}{% endif %}] {{ alertname }} {% if groupLabels | length > 0 %}({{ groupLabels|join(", ") }}){% endif %}"""  # noqa
 
 web_message = """\
-{%- set annotations = payload.commonAnnotations.copy() -%}
+{%- set annotations = payload.get("commonAnnotations", {}).copy() -%}
+{%- set groupLabels = payload.get("groupLabels", {}) -%}
+{%- set commonLabels = payload.get("commonLabels", {}) -%}
+{%- set severity = groupLabels.severity -%}
 
-{% set severity = payload.groupLabels.severity -%}
 {% if severity %}
 {%- set severity_emoji = {"critical": ":rotating_light:", "warning": ":warning:" }[severity] | default(":question:") -%}
 Severity: {{ severity }} {{ severity_emoji }}
 {% endif %}
 
-{%- set status = payload.status | default("Unknown") %}
+{%- set status = payload.get("status", "Unknown") %}
 {%- set status_emoji = {"firing": ":fire:", "resolved": ":white_check_mark:"}[status] | default(":warning:") %}
 Status: {{ status }} {{ status_emoji }} (on the source)
-{% if status == "firing" %}
+{% if status == "firing" and payload.numFiring %}
```

Comment:
should we add check for numResolved?

Reply:
I'm checking the numFiring here to not to render empty ':', since alerts from migrated legacy alertmanager have no numFiring field. I'm still thinking how to do it better.